pkg/iac/scanners/terraform/parser/parser.go

```diff
@@ -201,45 +172,61 @@ func (p *Parser) ParseFS(ctx context.Context, dir string) error {
 var ErrNoFiles = errors.New("no files found")
 
 func (p *Parser) Load(ctx context.Context) (*evaluator, error) {
-	p.debug.Log("Evaluating module...")
+	p.logger.Debug("Loading module", log.String("module", p.moduleName))
 
 	if len(p.files) == 0 {
-		p.debug.Log("No files found, nothing to do.")
+		p.logger.Info("No files found, nothing to do.")
 		return nil, ErrNoFiles
 	}
 
 	blocks, ignores, err := p.readBlocks(p.files)
 	if err != nil {
 		return nil, err
 	}
-	p.debug.Log("Read %d block(s) and %d ignore(s) for module '%s' (%d file[s])...", len(blocks), len(ignores), p.moduleName, len(p.files))
+	p.logger.Debug("Read block(s) and ignore(s)",
+		log.Int("blocks", len(blocks)), log.Int("ignores", len(ignores)))
 
 	var inputVars map[string]cty.Value
 	if p.moduleBlock != nil {
 		inputVars = p.moduleBlock.Values().AsValueMap()
-		p.debug.Log("Added %d input variables from module definition.", len(inputVars))
+		p.logger.Debug("Added input variables from module definition",
+			log.Int("count", len(inputVars)))
 	} else {
 		inputVars, err = loadTFVars(p.configsFS, p.tfvarsPaths)
 		if err != nil {
 			return nil, err
 		}
-		p.debug.Log("Added %d variables from tfvars.", len(inputVars))
+		p.logger.Debug("Added input variables from tfvars", log.Int("count", len(inputVars)))
+
+		for _, varBlock := range blocks.OfType("variable") {
+			if varBlock.GetAttribute("default") == nil {
+				if _, ok := inputVars[varBlock.TypeLabel()]; !ok {
+					p.logger.Warn(
+						"Variable was not found in the environment or variable files. Evaluating may not work correctly.",
+						log.String("variable", varBlock.TypeLabel()),
+					)
+				}
+			}
+		}
```

Comment:
Is there a test for this?

Reply:
Done https://github.com/aquasecurity/trivy/pull/7295/commits/1cb026394aba1ae41fb48de1e6d1daf71c7ab17b
Also changed the wording of the logs to mean missing values rather than missing variables/parameters.